index.js

```diff
@@ -1,33 +1,59 @@
 const fetch = require('node-fetch'); // Automatically excluded in browser bundles
 
-async function api(endpoint, token) {
-	token = token ? `&access_token=${token}` : '';
-	const response = await fetch(`https://api.github.com/repos/${endpoint}${token}`);
+// Matches '/<re/po>/tree/<ref>/<dir>'
+const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
+
+function parseResource(res) {
+	if (typeof res === 'string') {
+		try {
+			const parsedUrl = new URL(res);
+			res = {api: 'https://api.github.com'};
+			[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname);
+			if (parsedUrl.hostname !== 'github.com') {
```

Comment:
I'd rather see a clearer:
```suggestion
			if (parsedUrl.hostname === 'github.com') {
				res.api = 'https://api.github.com';
			} else {
```